### DISCUSSION TITLE
Can HEVC videos be played in PhotoPrism?

### DISCUSSION THREAD

**Discussion by masterwishx** [photoprism/photoprism#2210] [Q&A]
i Have some videos in HEVC (h.256) in Photoprism and they cant start playing in Chrome and Firefox .
is Photoprism cant play them or its browsers Problem ?

**Comment by lastzero** [comment]
Did you check if ffmpeg (the version Debian ships) supports it? Did you check your logs?

**Reply by lastzero** [reply]
Oh HEVC not NEVC.... yes should work, but requires transcoding. See our Docs and Checklists.

**Comment by masterwishx** [comment]
im using last stable version Build 220302-0059f429-Linux-AMD64
i checked log i have : 

![image](https://user-images.githubusercontent.com/28630321/161304570-abfd99db-f39c-486a-8a98-5e3038e5a284.png)

then i changed PHOTOPRISM_FFMPEG_ENCODER to intel from  h264_qsv

![image](https://user-images.githubusercontent.com/28630321/161308282-25f79b67-ac89-4645-b124-81ac1fd9a85f.png)

Also i have Unsupported GPU: "vfio-pci" in log on first start . but on vfio i have gt620 and intel iGPU Driver  installed in Unraid .

![image](https://user-images.githubusercontent.com/28630321/161308543-2db1285e-f780-47f4-9fcf-edf173c7d77b.png)

**Reply by lastzero** [reply]
Getting Hardware Transcoding to work is a completely different story than general h265 support. Does it work with the standard software encoder?

**Reply by masterwishx** [reply]
Do you mean if i use "software" instead of "intel" or "h264_qsv"

**Comment by masterwishx** [comment]
you mean h.264 ?
Yes its working

**Reply by masterwishx** [reply]
(HVC1) HEVC h.265 videos not playing , h.264 is working . 

 have CPU i5 4440i , i changed to **"intel"** from **"h264_qsv"** like in new docker-compose.yml , and i think same driver is installing ..
i also have some warning in ffmpeg in console:
**" WARNING: library configuration mismatch "**

![image](https://user-images.githubusercontent.com/28630321/161370941-4af54334-931c-4ab9-80f6-0fb2effda3a6.png)

**Comment by masterwishx** [comment]
i found my CPU (Haswell) isn't capable of transcoding HEVC (h265) : [Click](https://en.wikipedia.org/wiki/Intel_Quick_Sync_Video#Hardware_decoding_and_encoding)
So should i use "software" instead of "intel" or "h264_qsv"

**Comment by masterwishx** [comment]
maybe i can use "software" only for HEVC ?

**Reply by lastzero** [reply]
Decoding happens via software in any case, so probably it's broken for other video formats as well. Note H.264 does not require transcoding, thus it will always work.

**Reply by masterwishx** [reply]
Sorry im little confused what should i do to play both videos h.264 (working)  and h.265? 
i dont need PHOTOPRISM_FFMPEG_ENCODER at all for playing videos?
i can use PHOTOPRISM_FFMPEG_ENCODER: "intel" for my i5 4440 CPU (Haswell) to encode only ?
where to look for solving the Problem ?

**Reply by lastzero** [reply]
Browsers play AVC H.264 natively; you don't need transcoding for it. For any other codec, such as H.265, you need a working transcoder. The software encoder is the easiest to use, as there are no kernel modules to install and no system devices to share. Perhaps a community member can help with this, as we are very busy at the moment and you may need an hour of remote assistance to get to know your environment and set everything up properly. At the moment, hardware transcoding is only suitable for advanced users.

**Reply by masterwishx** [reply]
I'm sorry I'm know you very busy and I'm also waiting mostly wanted future that in development now, so no problem I will find  best choice for use PhotoPrism in my case.
Thanks
